configuration/config2/XMLEnvironmentMgr.cpp

```diff
@@ -42,7 +42,7 @@ bool XMLEnvironmentMgr::doLoadEnvironment(std::istream &in)
         if (rootName == m_pSchema->getProperty("name"))
         {
             m_pRootNode = std::make_shared<EnvironmentNode>(m_pSchema, rootName);
-            m_pRootNode->setId(".");
+            m_pRootNode->setId("0");
```

Comment:
since 0 has a special meaning I would #def 0 to a meaningful name

Reply:
Not sure what you mean. The next PR for HPCC-19216 makes a change in that the root node ID is not always fixed to "0", so this code changes. The latest changes for this PR includes the root node ID in the return of the load environment call (set to "0" until the newest changes are brought in)

Reply:
#define ROOT_NODE_IDENTIFIER "0", or something similar to that 
then m_pRootNode->setId(ROOT_NODE_IDENTIFIER); 
and if there's a getNodebyId(ROOT_NODE_IDENTIFIER); rather than using "0" literal.

Reply:
In the next PR, the root node will not always be "0". The root node will be returned to the UI when the environment is loaded.